lib/OpenQA/FakeApp.pm

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::FakeApp;
+use Mojo::Log;
+use Mojo::Home;
+use strict;
+use warnings;
+use Mojo::Base -base;
+use Sys::Hostname;
+use File::Spec::Functions qw/catfile/;
+
+has config => sub { {} };
+
+has log => sub {
+    my ($self) = @_;
+    my $logger = Mojo::Log->new;
```

Comment:
I meant, you can just:

```perl
has log => sub { Mojo::Log->new(handle => \*STDOUT, level=> "debug" ); }
```

and later you can call `setup_log` to set custom configurations (on top, overriding defaults when necessary)

Reply:
Unless i'm seeing it wrong, If i setup the handle, and then later if later i set up the path, the handle will not change.

`
$log = Mojo::Log->new(handle => \*STDOUT, level=> "debug" ); 
$log->path('/path/to/file')
`
won't log the output to "/path/to/file"

Reply:
yes, you need to setup also the handle pointing to a Mojo::File class, so that would be sufficient:

```perl
use Mojo::File 'path';
$self->log->handle(path($path)->open('>>'))
```